src/classes/RD_RecurringDonations_Opp_TDTM.cls

```diff
@@ -56,13 +56,20 @@ public class RD_RecurringDonations_Opp_TDTM extends TDTM_Runnable {
             for (SObject sobj : newlist) {
                 Opportunity o = (Opportunity)sobj;
                 //does it have a recurring donation reference?
-                //has it changed closed state or amount?
+                //has it changed closed state, amount, or recurring donation changed?
                 Opportunity oppOld = (Opportunity)oldlist[i];
-                if (o.npe03__Recurring_Donation__c != null && (o.isClosed != oppOld.isClosed || o.Amount != oppOld.Amount)){
+                if (o.isClosed != oppOld.isClosed 
+                    || o.Amount != oppOld.Amount
+                    || o.npe03__Recurring_Donation__c != oppOld.npe03__Recurring_Donation__c){
+                    // If the Recurring Donation has changed, also update the Recurring Donation 
+                    // associated with the old (previous) version of the Opportunity
+                    if(o.npe03__Recurring_Donation__c != oppOld.npe03__Recurring_Donation__c) {
+                        rdIds.add(oppOld.npe03__Recurring_Donation__c);
+                    }
                     rdIds.add(o.npe03__Recurring_Donation__c);
```

Comment:
Same here, setting the Opp's RD to null, removed the Opp from the RD and updated its values. I'll test again.

Thanks!